corgi/core/models.py

```diff
@@ -1628,7 +1640,11 @@ def save_product_taxonomy(
         self.productversions.add(*product_pks_dict["productversions"])
         self.productstreams.add(*product_pks_dict["productstreams"])
         self.productvariants.add(*product_pks_dict["productvariants"])
```

Comment:
The problem is with UPSTREAM components, because we call Component.save_product_taxonomy() via SoftwareBuild.save_product_taxonomy() we are adding product relationships for that build only. UPSTREAM components can and are used in multiple builds, if we replaced the relations of these components every time they would only reflect the last builds relations.

Take for example a GOLANG component (x/text) bundled by the "cri-o" RPM, but also build into the "etcd" Container Image. If we called the "cri-o" RPM save_product_taxonomy first the (x/text) component would reflect the products of the cri-o RPM. However later if we called the "etcd" Container Image build save_product_taxonomy function (x/text) would not only reflect the products of the "etc" Container Image whereas it should reflect both.